src/_articles/archive/zones.md

```diff
@@ -152,33 +154,51 @@ Code might execute in other nested child zones as well,
 but at a minimum it always runs in the root zone.
 
 
-## Handling asynchronous errors
+## Handling uncaught errors
 
 One of the most used features of zones is
-their ability to handle uncaught errors
-in asynchronously executing code.
-The concept is similar to a try-catch in synchronous code.
-
-An _uncaught error_ is often caused by some code using `throw`
-to raise an exception that is not handled by a `catch` statement.
-Another way to produce an uncaught error is
-to call `new Future.error()` or
-a Completer's `completeError()` method.
-
-Use the `onError` argument to `runZoned()` to
-install a _zoned error handler_—an asynchronous error handler
-that's invoked for every uncaught error in the zone.
-For example:
+their ability to catch and handle uncaught errors,
+both synchronous and asynchronous.
+An _uncaught error_ is often caused by code using `throw`
+to raise an exception without an accompanying `catch`
+statement to handle it.
+
+In `async` functions, errors usually end up as the result of a Future.
+Futures allow you to handle errors where you `await` the future.
+Sometimes an error escapes, or a future is not `await`-ed,
+in which case an error becomes an uncaught error.
+Uncaught errors can crash your program.
```

Comment:
Oh I think this is unclear. "errors usually end up as the result of a Future." doesn't mean the Future is causing errors, it means that when an error happens in async code, it is returned as a result _to_ a Future. Is that what you meant by "qualifier"?

I could reword it:
> In `async` functions, errors are usually returned as a result to a Future.

Reply:
**reply:** That's definitely better. I would adjust, though, to clarify what has agency here. What returns an error? The function, then maybe:
```markdown
Async functions tend to return errors due to a Future.
```

I would get away from `usually` or adverbs in general. Find a better verb, because something like `usually` begs for `what about unusually`?

Reply:
I think your suggestion is missing the meaning again, because of `due to`. The future is not causing anything here, it's just what async function results _are_. It's like a return type almost. 

The sentence is also not about async functions, it's about the errors: what happens with uncaught errors in aysnc functions. 

I actually think the original way the sentence is written is kind of the only good way to capture the point, so I'm going to change `usually` to `tend to` but otherwise leave it as is. I think the confusing part here is that I separated a continuous thought with a new paragraph, so I undid that. Maybe with the previous sentence connected, it's clearer:
> An _uncaught error_ is often caused by code using `throw`
to raise an exception without an accompanying `catch`
statement to handle it.
In `async` functions, these errors tend to end up as the result of a Future.

Reply:
Ahh wait I get it now, my interpretation was incorrect. It's "Uncaught errors in async functions tend to happen when attempting to handle errors with Futures" ..... I think. 

I think I can actually rewrite that in the sentence form you suggested. And it's not about the paragraphs being separated, they actually should be separate.... I'll re-request your review on the next commit